src/App/Pages/Accounts/LoginSsoPageViewModel.cs

```diff
@@ -65,81 +69,96 @@ public async Task InitAsync()
-                      "client_id=" + _platformUtilsService.GetClientType().GetString() + "&" +
-                      "redirect_uri=" + Uri.EscapeDataString(redirectUri) + "&" +
-                      "response_type=code&scope=api%20offline_access&" +
-                      "state=" + state + "&code_challenge=" + codeChallenge + "&" +
-                      "code_challenge_method=S256&response_mode=query&" +
-                      "domain_hint=" + Uri.EscapeDataString(OrgIdentifier) + "&" +
-                      "ssoToken=" + Uri.EscapeDataString(ssoToken);
+                var codeVerifier = await _passwordGenerationService.GeneratePasswordAsync(passwordOptions);
+                var codeVerifierHash = await _cryptoFunctionService.HashAsync(codeVerifier, CryptoHashAlgorithm.Sha256);
+                var codeChallenge = CoreHelpers.Base64UrlEncode(codeVerifierHash);
+
+                var state = await _passwordGenerationService.GeneratePasswordAsync(passwordOptions);
+
+                var redirectUri = "bitwarden://sso-callback";
+
+                var url = _apiService.IdentityBaseUrl + "/connect/authorize?" +
+                          "client_id=" + _platformUtilsService.GetClientType().GetString() + "&" +
+                          "redirect_uri=" + Uri.EscapeDataString(redirectUri) + "&" +
+                          "response_type=code&scope=api%20offline_access&" +
+                          "state=" + state + "&code_challenge=" + codeChallenge + "&" +
+                          "code_challenge_method=S256&response_mode=query&" +
+                          "domain_hint=" + Uri.EscapeDataString(OrgIdentifier) + "&" +
+                          "ssoToken=" + Uri.EscapeDataString(ssoToken);
+
+                WebAuthenticatorResult authResult = null;
 
-            WebAuthenticatorResult authResult = null;
-            try
-            {
                 authResult = await WebAuthenticator.AuthenticateAsync(new Uri(url),
                     new Uri(redirectUri));
+
+
+                var code = GetResultCode(authResult, state);
+                if (!string.IsNullOrEmpty(code))
+                {
+                    await LogIn(code, codeVerifier, redirectUri, OrgIdentifier);
+                }
+                else
+                {
+                    await _deviceActionService.HideLoadingAsync();
+                    await _platformUtilsService.ShowDialogAsync(AppResources.LoginSsoError,
+                        AppResources.AnErrorHasOccurred);
+                }
             }
-            catch (TaskCanceledException)
+            catch (ApiException e)
             {
-                // user canceled
+                _logger.Exception(e);
                 await _deviceActionService.HideLoadingAsync();
+                await _platformUtilsService.ShowDialogAsync(
+                    (e?.Error != null ? e.Error.GetSingleMessage() : AppResources.LoginSsoError),
```

Comment:
unless I'm missing something parenthesis are not needed here and it can be written as:
`e?.Error?.GetSingleMessage() ?? AppResources.LoginSsoError`